src/components/Header/Header.js

```diff
@@ -34,7 +34,7 @@ const StyledHeader = styled.div`
 
 const Header = () => {
     const {user, login, logout} = useContext(UserContext)
-    const {setNotification} = useContext(NotificationContext)
+    const {setNotification} = useContext(UserNotificationContext)
```

Comment:
I also am not sure it's necessary for the `Header` to be the one to emit the notification. Why not have the User context have access to the notifications? Downside is that you'd have to ensure proper nesting.

Reply:
It is not mandatory for every consumer of User context to have Notifications access. UserNotifications is for those who need it. For checkout context, it only needs user data but for settings context, it needs user and notifications. Depends on the use case.

Reply:
But you add complexity by creating one-off context providers rather than wrapping your app in context. As for performance I'm not sure which is better.

Reply:
Since we are using the same memory reference of the contexts of User and Notification inside UserNotification, it won't be a memory issue. We can use n number of such combinations.